### PR TITLE
FIX: Create a 3rd party from the self-subscription membership form

### LINKED ISSUE #21095: No thirdparty created while using the public member creation form
### Bug

When new member create a new adherent with the public form and pay his or her subscription (by stripe), no thirdparty is created, even if the hidden option ADHERENT_DEFAULT_CREATE_THIRDPARTY is set to 1.

So there's a facture and there's a bank transaction but no thirdparty linked. So it may be problematic for accountacy.

### Environment Version

14.0.5

### Environment PHP

PHP 7.4

### Environment Database

MySQL

### Environment URL(s)

htdocs/public/members/new.php

### Expected and actual behavior
New memeber is created
Payement is OK
Subscription is registered
Bank transaction is notified
But there is no thirdparty linked to the operation since there is no thirdparty created and linked to the member

### PR DESCRIPTION
**Pull request by ymollard**
# FIX https://github.com/Dolibarr/dolibarr/issues/21095 Create a 3rd party from the self-subscription membership form

Only when ADHERENT_DEFAULT_CREATE_THIRDPARTY = 1

### CODE REVIEW

**Comment by daraelmin** [comment]
Thx

File: htdocs/societe/class/societe.class.php [hunk 1]
```diff
@@ -3904,12 +3904,21 @@ public function create_from_member(Adherent $member, $socname = '', $socalias =
 		global $conf, $user, $langs;
 
 		dol_syslog(get_class($this)."::create_from_member", LOG_DEBUG);
+		$fullname = $member->getFullName($langs);
 
-		$name = $socname ? $socname : $member->societe;
-		if (empty($name)) {
-			$name = $member->getFullName($langs);
+		if ($member->morphy == 'mor') {
+			$socname = $member->company? $member->company : $member->societe;
```
Comment (eldy) [inline htdocs/societe/class/societe.class.php:3910]: If $socname or $socalias was forced by caller, here you erased the value.
I suggest to implement the logic to define socname and socalias in this method only if input var are not forced by caller, for compatibility and to keep the use of existing parameters.
So you should add a test` if (empty($socname))`

File: htdocs/societe/class/societe.class.php [hunk 2]
```diff
@@ -3904,12 +3904,21 @@ public function create_from_member(Adherent $member, $socname = '', $socalias =
...
+			if (!empty($fullname) && empty($socalias)) {
+				$socalias = $fullname;
+			}
+		} elseif (empty($socname) && $member->morphy == 'phy') {
+			$socname = $fullname;
```
Comment (eldy) [inline htdocs/societe/class/societe.class.php:3915]: Same here: erase $socname if not already forced.
